stan/math/fwd/mat/fun/inverse.hpp

```diff
@@ -15,12 +15,13 @@ namespace math {
 template <typename T, int R, int C>
 inline Eigen::Matrix<fvar<T>, R, C> inverse(
     const Eigen::Matrix<fvar<T>, R, C>& m) {
+  check_nonempty("inverse", "m", m);
```

Comment:
@mcol you are working on these a lot. Just double checking with you if this is what we want or would returning an empty matrix be better. I do think we should check that the matrix is square before the nonempty check.

Reply:
For inverse we always throw if it's empty, so the order of the checks doesn't quite matter (see #1464 for example).

Reply:
Sweet. All good then. Thanks @mcol

Reply:
Looks like you've figured it out, but that function threw on non empty anyway. Inverse<fwd<T> > calls inverse<T>, which eventually terminates at either inverse<prim> or inverse<var>. Prim and var threw on non-empty, so fwd did too. This change just makes it explicit.